examples/step-98/step-98.cc

```diff
@@ -0,0 +1,1106 @@
+   * (ghost value updates and result compression) implicitly.
+   */
+  template <int dim, int fe_degree, typename number>
+  void LaplaceOperator<dim, fe_degree, number>::apply_add(
+    VectorType       &dst,
+    const VectorType &src) const
+  {
+    this->data->cell_loop(&LaplaceOperator::local_apply, this, dst, src);
+  }
+
+
+
+  /**
+   * @brief Implements the inverse of the local patch operator using tensor
+   * product techniques.
+   *
+   * This class computes and applies the inverse of the Laplace operator
+   * assembled on a single patch (typically consisting of $2^d$ cells). It
+   * exploits the tensor product structure of the finite element basis and
+   * quadrature on the reference patch to efficiently compute the inverse.
+   *
+   * The patch operator is assumed to be a sum of Kronecker products involving
+   * 1D mass and stiffness matrices. This class pre-computes these 1D matrices
+   * and uses the TensorProductMatrixSymmetricSum class to apply the inverse.
+   *
+   */
+  template <int dim, int fe_degree, typename number, typename VectorizedValues>
+  class PatchTensorInverse
+  {
+  public:
+    //  using value_type = number;
+
+
+    /**
+     * @brief The size (number of rows/columns) of the 1D matrices that form
+     * the tensor product patch matrix. For a patch of $2^d$ cells with DGQ
+     * elements of degree `fe_degree`, this corresponds to the number of unique
+     * 1D basis functions on the two adjacent reference cells along one axis.
+     */
+    const constexpr static unsigned int n_rows_1d = 2 * fe_degree - 1;
+
+    /**
+     * @brief Clear internal data structures.
+     */
+    void clear()
+    {}
+
+    /**
+     * @brief Initialize the patch inverse operator.
+     *
+     * This function computes the 1D mass and stiffness matrices on the
+     * reference cell, scales them according to the mapping information
+     * obtained from the MatrixFree object, assembles the patch matrices
+     * (patch_mass, patch_laplace) by combining contributions from adjacent
+     * cells, and initializes the TensorProductMatrixSymmetricSum object used
+     * for applying the inverse.
+     *
+     * @param mf A shared pointer to the MatrixFree object containing geometric
+     *           and finite element information. Assumes a uniform grid patch.
+     */
+    void initialize(std::shared_ptr<const MatrixFree<dim, number>> mf);
+
+    /**
+     * @brief Apply the inverse of the patch matrix to a vector.
+     *
+     * Solves the linear system `patch_matrix * dst = src` for `dst`.
+     *
+     * @tparam VectorType The type of the source and destination vectors
+     *                    (typically AlignedVector<VectorizedValues>).
+     * @param dst The destination vector (solution).
+     * @param src The source vector (right-hand side).
+     */
+    template <typename VectorType>
+    void vmult(VectorType &dst, const VectorType &src) const;
```

Comment:
I would avoid to call the template argument `VectorType` here, because here we are addressing a local solver on a patch, whereas the same name `VectorType` is used in the `LaplaceOperator` class to refer to the global vector. Would it make sense to call it `PatchVectorType` in your opinion?